README.md

```diff
@@ -1,105 +1,257 @@
+    ```bash
+    make module-image IMG_REGISTRY=localhost:5001/unsigned/operator-images IMG=localhost:5001/serverless-manager-dev-local:0.0.1
+    ```
 
-```sh
-make run
-```
+5. Build and push the Serverless module
 
-**NOTE:** You can also run this in one step by running: `make install run`
+    ```bash
+    make module-build IMG=k3d-kyma-registry:5001/serverless-manager-dev-local:0.0.1 MODULE_REGISTRY=localhost:5001/unsigned
+    ```
 
-### Modifying the API definitions
-If you are editing the API definitions, generate the manifests such as CRs or CRDs using:
+6. Verify if the module and the manager's image are pushed to the local registry
 
-```sh
-make manifests
-```
+    ```bash
+    curl localhost:5001/v2/_catalog
+    ```
 
-**NOTE:** Run `make --help` for more information on all potential `make` targets
+    The example result should look like this:
 
-More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
+    ```json
+    {"repositories":["serverless-manager-dev-local","unsigned/component-descriptors/kyma.project.io/module/serverless"]}
+    ```
 
-## License
+7. Patch CoreDNS resource on the Kyma cluster
 
-Copyright 2022.
+    ```bash
+    bash hack/get_kyma_localhost_registry_name.sh k3d-kyma-registry
+    ```
 
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
+8. Inspect the generated module template
 
-    http://www.apache.org/licenses/LICENSE-2.0
+    > **NOTE:** The following sub-steps are temporary workarounds.
 
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
+    Edit the `template.yaml` file and:
 
+    - change `target` to `control-plane`
+
+    > **NOTE:** This is only required in the single cluster mode.
+
+    ```yaml
+    spec:
+        target: control-plane
+    ```
+
+    - change the existing repository context in `spec.descriptor.component`:  
+    
+    > **NOTE:** Because Pods inside the k3d cluster use the docker-internal port of the registry, it tries to resolve the registry against port 5000 instead of 5001. K3d has registry aliases, but `module-manager` is not part of k3d and thus does not know how to properly alias `k3d-kyma-registry.localhost:5001`
+
+    ```yaml
+    repositoryContexts:                                                                           
+      - baseUrl: k3d-kyma-registry.localhost:5000/unsigned
+        componentNameMapping: urlPath                                                               
+        type: ociRegistry
+    ```
```

Comment:
```suggestion
    ```yaml
    repositoryContexts:                                                                           
      - baseUrl: k3d-kyma-registry.localhost:5000/unsigned
        componentNameMapping: urlPath                                                               
        type: ociRegistry
    ```

    > **NOTE:** Because Pods inside the k3d cluster use the docker-internal port of the registry, it tries to resolve the registry against port 5000 instead of 5001. K3d has registry aliases, but `module-manager` is not part of k3d and thus does not know how to properly alias `k3d-kyma-registry.localhost:5001`
```